File: final_back/app/Models/Brand.ts
import { BaseModel, HasMany, column, hasMany } from '@ioc:Adonis/Lucid/Orm'
import Product from './Product'

export default class Brand extends BaseModel {
  public static table= "brands"
  @column({ isPrimary: true })
  public id: number

  @column ({serializeAs: "name"})
  public name: string

  @column ({serializeAs: "image"})
  public image: string

  @column ({serializeAs: "description"})
  public description: string
  
  @hasMany(()=> Product)
  public product: HasMany<typeof Product>
}

File: final_project/lib/models/brand.dart
class Brand {
  int id;
  String name;
  String image;
  String description;

  Brand(this.id, this.name, this.image, this.description);

  factory Brand.fromJson(json) {
    return Brand(
      json["id"] ?? 0,
      json["name"] ?? "",
      json["image"] ?? "",
      json["description"] ?? "",
    );
  }
  Map<String, dynamic> toJson() {
    return {
      "id": id,
      "name": name,
      "image": image,
      "description": description,
    };
  }
}
